bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/graphics/SVGUtil.java

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse
+ * Public License 2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors: Vector Informatik GmbH - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.swt.graphics;
+
+import java.io.*;
+import java.nio.charset.*;
+
+/**
+ * Utility class for handling SVG-related operations.
+ *
+ * @since 3.129
+ */
+public class SVGUtil {
+
+	/**
+	 * Determines whether the given {@link InputStream} contains a SVG file.
+	 *
+	 * @param data byte array to check.
+	 * @return {@code true} if the input stream contains SVG content; {@code false}
+	 *         otherwise.
+	 * @throws IOException              if an error occurs while reading the stream.
+	 * @throws IllegalArgumentException if the input stream is {@code null}.
+	 */
+	public static boolean isSVGFile(byte[] data) throws IOException {
+		String content = new String(data, 0, Math.min(data.length, 512), StandardCharsets.UTF_8);
+		return content.contains("<svg");
+	}
+
+	/**
+	 * Determines whether the given {@link InputStream} contains a SVG file.
+	 *
+	 * @param inputStream the input stream to check.
+	 * @return {@code true} if the input stream contains SVG content; {@code false}
+	 *         otherwise.
+	 * @throws IOException              if an error occurs while reading the stream.
+	 * @throws IllegalArgumentException if the input stream is {@code null}.
+	 */
+	public static boolean isSVGFile(InputStream inputStream) throws IOException {
+		if (inputStream == null) {
+			throw new IllegalArgumentException("InputStream cannot be null");
+		}
+		byte[] data = inputStream.readNBytes(512);
+		return isSVGFile(data);
```

Comment:
This will corrupt the input stream, so it is quite useless method as afterwards the stream is unusable.

You should use mark / reset so the stream is in the same state as before (what would requires callers to use a BufferedStream).

Reply:
The caller is SWT code so this can be made work for sure ;-)

Otherwise as mentioned I don't see any benefit in providing a stream variant at all if the stream is unusable after that.

Reply:
The streams come from two sources: 
- For Standard-Icons or graphically customized icons these are always FileInputStreams that can be wrapped to a BufferedInputStream. 
- For Composite-Icons they are already BufferedInputStreams. 

The only problem that I have while testing is that the `SVGLoader.load` method of JSVG closes the stream which prevents a reset after the call. @weisJ do you know why this happens?

Reply:
JSVG wraps the given InputStream, hence also closes it. This might close the wrapped stream as well.

Reply:
> JSVG wraps the given InputStream, hence also closes it. This might close the wrapped stream as well.

yes it does. I will try to find another solution to this.

Reply:
Just wrap your stream in a `FilteredInputStream` and override close method to do nothing.

Reply:
> I searched for a way to solve this differently but wasn't successful. If someone has an idea I would be grateful.

Using a `PushbackInputStream` could probably help here and you could still consider streams that support marks:
```
	if (stream.markSupported()) {
		stream.mark(1);
		int firstByte = stream.read();
		if (firstByte != '<') {
			// TODO: handle not an SVG, maybe return
		}
		stream.reset();
	} else {
		PushbackInputStream pushbackStream=new PushbackInputStream(stream,1);
		int firstByte = stream.read();
		if (firstByte != '<') {
			// TODO: handle not an SVG, maybe return
		}
		pushbackStream.unread(firstByte);
		stream = pushbackStream;
	}
	// ... rasterize the SVG
```

Reply:
Or yes, it could also be sufficient to just wrap the given `InputStream` into a `BufferedInputStream` if mark `markSupported()` returns false, before it is passed to a method that might read a few bytes to test if the content is a SVG.

Reply:
@HannesWell this will only work if you "own" the stream, but I though the problem was more that JSVG closes the stream.

In anyway we should not complicate this too much this is not something I would assume to become a popular API with many callers, so we can simply make some assertions / requirements (e.g if one want to be explicitly require a `PushbackInputStream`).

Reply:
> Just wrap your stream in a `FilteredInputStream` and override close method to do nothing.

This worked for me. See b5365d82069f08efd7ae04da048a65ea2271b1ec for the commit.